### PR TITLE
Add ability for the spatial mesh to move with the user

### PR DESCRIPTION
**Pull request by keveleigh**
Adds a helper script to update the mesh observer's origin to move with the user.

### CODE REVIEW

**Review by yl-msft — APPROVED** [review]

File: BasicSample/Assets/ARMesh/Scripts/MeshFollowUser.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+using UnityEngine;
+using UnityEngine.XR.ARFoundation;
+
+namespace Microsoft.MixedReality.OpenXR.BasicSample
+{
+    /// <summary>
+    /// Updates the mesh bounding volume every frame to follow the user's head.
+    /// </summary>
+    public class MeshFollowUser : MonoBehaviour
```
Comment (yl-msft) [inline BasicSample/Assets/ARMesh/Scripts/MeshFollowUser.cs:12]: nit:  consider to rename.  Considering the typical usage is to put this script side by side with the "ARMeshManager", How about calling this "ARMeshUpdateFollowsUser"?

File: BasicSample/Assets/ARMesh/Scripts/MeshFollowUser.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        [SerializeField]
+        private ARMeshManager meshManager = null;
+
+        [SerializeField]
+        private Vector3 boundingExtents = Vector3.one * 3;
+
+        private void Update()
+        {
+            meshManager.subsystem.SetBoundingVolume(Camera.main.transform.position, boundingExtents);
```
Comment (yl-msft) [inline BasicSample/Assets/ARMesh/Scripts/MeshFollowUser.cs:22]: Also consider to add a logic to the "meshManager" here.   If it's null, look for the ARMeshManager on the current game object.  So user can easily drop this to the same game object and get it working.

**Review by yl-msft — APPROVED** [review]

File: BasicSample/Assets/ARMesh/Scripts/MeshFollowUser.cs (hunk 1 above)
Comment (keveleigh) [reply BasicSample/Assets/ARMesh/Scripts/MeshFollowUser.cs:12]: Updated!

File: BasicSample/Assets/ARMesh/Scripts/MeshFollowUser.cs (hunk 2 above)
Comment (keveleigh) [reply BasicSample/Assets/ARMesh/Scripts/MeshFollowUser.cs:22]: Added!

**Pull request merged**